EIPS/eip-8888.md

```diff
@@ -0,0 +1,194 @@
+| `mixhash`       | Hex-encoded mix hash.                                           |
+| `coinbase`      | Hex-encoded address.                                            |
+
+### `config` Object
+
+The `config` object contains hardfork activation block numbers and fork configurations. Known keys include:
+
+| Field                     | Description                                                                      |
+|---------------------------|----------------------------------------------------------------------------------|
+| `chainId`                 | unique identifier for the blockchain.                                            |
+| `<hardfork(Block\|Time)>` | block height or timestamp to activate the named hardfork.                        |
+| `terminalTotalDifficulty` | difficulty after which to switch from PoW to PoS.                                |
+| `depositContractAddress`  | Ethereum address for the deposit contract                                        |
+| `blobSchedule`            | Map of hardforks and their EIP-4844 DAS [configuration parameters](eip-7840.md). |
+
+### `blobSchedule` Object
+
+| Field                   | Description                                  |
+|-------------------------|----------------------------------------------|
+| `target`                | desired number of blobs to include per block |
+| `max`                   | maximum number of blobs to include per block |
+| `baseFeeUpdateFraction` | input to pricing formula per EIP-4844        |
+
+### `alloc` Object
+
+The `alloc` field defines the initial state at genesis. It maps addresses (as lowercase hex strings) to the following object:
+
+| Field          | Description                                     |
+|----------------|-------------------------------------------------|
+| `balance`      | decimal balance in wei.                         |
+| `code`         | Hex-encoded EVM bytecode.                       |
+| `nonce`        | decimal value.                                  |
+| `storage`      | Key-value hex map representing initial storage. |
+
+
+
+JSON Schema
+
+
+```json
+
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$defs": {
+  	"hexOrDecimal48": {
+    		"type": "string",
+    		"pattern": "^(0x[0-9a-fA-F]{1,12}|[0-9]{1,12})$"
+  	},
+    "address": {
+      "type": "string",
+      "pattern":  "^0x[0-9a-fA-F]{40}$"
+    },
+    "hash": {
+      "type": "string",
+      "pattern": "^0x[0-9a-fA-F]{64}$"
+    }
+  },
+  "title": "Ethereum Genesis File",
+  "type": "object",
+  "required": ["alloc", "gasLimit", "difficulty"],
+  "properties": {
+    "config": {
+      "type": "object",
+      "properties": {
+        "chainId": { "type": "integer" },
+        "homesteadBlock": { "type": "integer" },
+        "daoForkBlock": { "type": "integer" },
+        "eip150Block": { "type": "integer" },
+        "eip155Block": { "type": "integer" },
+        "eip158Block": { "type": "integer" },
```

Comment:
We need to fix these fork names and not use the eip names. 

EIP150 is TangerineWhistle
EIP155 is SpuriousDragon
EIP158 is never shipped (but EIP161 is, and was shipped in SpuriousDragon).

So let's add `tangerineWhistleBlock` and `spuriousDragonBlock`

Reply:
They've been added, but didn't remove eip15*Block, since Geth needs 'em:

https://github.com/ethereum/go-ethereum/blob/master/params/config.go#L391

Reply:
Right, yes, I see. In this specific case I think we need to change the format. This format now introduces 3 forks (where there should be 2 -> 155/158 are split up), the name is incorrect (EIP158), and they are not indexed like others by their hardfork name.

Here is an excerpt from the mapper in hive:

```jq
    "eip150Block": env.HIVE_FORK_TANGERINE|to_int,
    "eip150Hash": env.HIVE_FORK_TANGERINE_HASH,
    "eip155Block": env.HIVE_FORK_SPURIOUS|to_int,
    "eip158Block": env.HIVE_FORK_SPURIOUS|to_int,
```

https://github.com/ethereum/hive/blob/874e7bc198afc224742706862ea40856a01de73b/clients/go-ethereum/mapper.jq#L42